src/com/carolinarollergirls/scoreboard/defaults/DefaultClockModel.java

```diff
@@ -321,6 +321,9 @@ public void setCountDirectionDown(boolean down) {
             Boolean last = new Boolean(countDown);
             countDown = down;
             scoreBoardChange(new ScoreBoardEvent(this, EVENT_DIRECTION, new Boolean(countDown), last));
+            if (last != down) {
+                setTime(getInvertedTime());
+            }
```

Comment:
In the same spirit of keeping `timeElapsed` constant, we should change the time of a countdown clock, when the maximum time is changed. (Most common use case by far for changing maximum time is changing halftime duration after it started.)

Reply:
Done